network/transport/types.go

```diff
@@ -211,10 +144,13 @@ func (s *NutsCommURL) UnmarshalJSON(bytes []byte) error {
 	if err := json.Unmarshal(bytes, &str); err != nil {
 		return errors.New("endpoint not a string")
 	}
-	endpoint, err := parseNutsCommAddress(str)
+	endpoint, err := core.ParsePublicURL(str)
```

Comment:
oh now I get it, but kinda confusing though (naming). But maybe a bit of code duplication won't hurt here?

If you're going to have a shared function for checking URLs, I'd make it more useful by adding `allowsSchemes` and `allowReservedHostnames` as parameters, then you can remove most of the new code in auth and in this file.

And the name is probably more like a "validate" function than "parse"?

Reply:
Added `allowedScemes` and `allowReserved` to the signature. I only expect a boolean or maybe an error with more details from a validate function, not the actual URL, so left at parse.